### DISCUSSION TITLE
Time differences of 1 hour

### DISCUSSION THREAD

**Discussion by fixmypcmike** [WinMerge/winmerge#2186] [Q&A]
Different decompression utilities for .zip files sometimes result in files with time differences of 1 hour, presumably depending on daylight savings time settings of the computer on which the zip was created and/or the computer on which the files are extracted.  Is there any option in Winmerge to ignore differences of 1 hour when comparing folders by Modified Date and Size?

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not have an option to ignore timestamp differences caused by Daylight Saving Time.

**Reply by melham01-DKC** [reply]
The 1-hr time difference between NTFS/FAT DST and STD time has been the biggest announce with WinMerge. I like WinMerge a lot but you need to include this feature.
The date, minutes and seconds are the same but the hour is different.  If you could have an option where you ignore the hour for date-time compare. that would cover 99% of all files, except for ones modified around midnight.

**Reply by boar24** [reply]
The hour should not be completely ignored, only if the difference is exactly one (1) hour should the files be considered identical.
And it should be configurable to the user if he wants the 1-hr time difference being ignored or not.

@fixmypcmike: it does not only affect zip files, but also e.g. jpg- and mp4-files.

**Comment by melham01-DKC** [comment]
Boar24,
 Agreed, that this feature (ignore time difference of one hour) should be
an option, just like "check only on size".

Case in point, we had a time change couple weeks back and my 14GB of backup
on a Flash drive had to be unnecessarily re-written because of the time
change.

Mike L.

**Comment by danicotra** [comment]
I think [**display filter line**](https://manual.winmerge.org/en/Filters.html#Filters_DisplayLineFilter) wasn't yet available at the time of OP writing... But now it is!

Right now, you can achieve what you need by typing `Ctrl+Shilf+L` and pasting what follows:
`fe!: Different and (LeftSize = RightSize) and (abs(LeftDate - RightDate) = 1h)`
this will exclude from result list every row having a couple of file with (same name and path/location and) same size but a time difference of _exactly_ 1 hour. (I personally think you should only need this when doing a "Date&Size only" comparison.)

P.S.
If you need to "aim" only at file with some specific extension, you can add, for instance, what follows after it:
` and Name recontains "\.(?:avif?|mp[34]|mov|m4a|jpe?g|png|webp|pdn|exe|epub|pdf|t?[gx]z|zip)$"`
 (that is a -surely incomplete- string listing which file extensions I usually want to "ignore" when the only difference is 1 hour)

P.P.S.
Also look at this: https://github.com/WinMerge/winmerge/issues/1175
